mat64/eigen.go

```diff
@@ -23,6 +29,90 @@ func symmetric(m *Dense) bool {
 	return true
 }
 
+// EigenSym is a type for creating and manipulating the Eigen decomposition of
+// symmetric matrices.
+type EigenSym struct {
+	vectorsComputed bool
+
+	values  []float64
+	vectors *Dense
+}
+
+// Factorize computes the Eigen decomposition of the symmetric matrix a.
+// The Eigen decomposition is defined as
+//  A = P * D * P^-1
+// where D is a diagonal matrix containing the eigenvalues of the matrix, and
+// P is a matrix of the eigenvectors of A. If the vectors input argument is
+// false, the eigenvectors are not computed.
+//
+// Factorize returns whether the decomposition succeeded. If the decomposition
+// failed, routines that require a successful factorization will panic.
+func (s *EigenSym) Factorize(a Symmetric, vectors bool) (ok bool) {
+	n := a.Symmetric()
+	sd := NewSymDense(n, nil)
+	sd.CopySym(a)
+
+	jobz := lapack.EigValueOnly
+	if vectors {
+		jobz = lapack.EigDecomp
+	}
+	w := make([]float64, n)
+	work := make([]float64, 1)
+	lapack64.Syev(jobz, sd.mat, w, work, -1)
+
+	work = make([]float64, int(work[0]))
+	ok = lapack64.Syev(jobz, sd.mat, w, work, len(work))
+	if !ok {
+		s.vectorsComputed = false
+		s.values = nil
+		s.vectors = nil
+		return false
+	}
+	s.vectorsComputed = vectors
+	s.values = w
+	s.vectors = NewDense(n, n, sd.mat.Data)
+	return true
+}
+
+// succFact returns true if the receiver contains a successful factorization.
+func (es *EigenSym) succFact() bool {
+	return len(es.values) != 0
+}
+
+// Values extracts the eigenvalues of the factorized matrix. If destination is
```

Comment:
Please unify how dst is referred to in this paragraph. At the moment it is "destination", "receiver", and "dst".